### ISSUE TITLE
server_shared_memory contains a string of platform-dependent size.

### ISSUE THREAD

**Issue by cmaughan** [supercollider/supercollider#4748]
The server_shared_memory structure in 
`https://github.com/supercollider/supercollider/blob/develop/common/server_shm.hpp`

Has a problem.  The size of the string object (not the size of the text), is different depending on platform.  Sometimes it is 32 bytes, sometimes 24 or 40.  There is no guarantee here of how much the string class reserves for its own purposes.  This means that a client trying to communicate with supercollider needs to reserve the _same_ number of bytes.  This is even different across debug/release versions.

In short, the shared memory should not contain a std::string type.  Perhaps a char buff[FIXED]...

**Comment by brianlheim** [comment]
thanks for the report! i wasn't aware this was something in use by third party clients. your solution of hardcoding the maximum buffer size sounds like a good solution to me. could you please submit a PR for that?

**Comment by cmaughan** [comment]
I'm not setup to build supercollider currently, but I will try to figure it out as soon as I get chance.  Sonic Pi uses this interface to draw the audio waveforms in its 'scope' feature.

**Comment by brianlheim** [comment]
@cmaughan sorry for the late reply.

> Is there an alternative/better way?

not one that i know of, but my knowledge of that part of the codebase is fairly shallow. your solution is probably the most expedient.

**Comment by brianlheim** [comment]
closing as stale. @cmaughan please submit a patch if you have time!

**Comment by oneturkmen** [comment]
@cmaughan I could try to set up the build and submit a patch. Do you have any tips on the size of FIXED for `char buff[FIXED]`?

**Comment by cmaughan** [comment]
Hi @oneturkmen Sorry for the delay.  If you are going to pursue this, let me know, otherwise I will take a stab at it before the end of this month.  The short answer is that the the buffer should be whatever size the string that Supercollider is putting in there is (I think it is predetermined). But if it varies in size, then anything reasonable should be OK.  I doubt it would impact anything if it was big, say 1024.
